### PR TITLE
Feature/project galaxy plugin

### PR DESCRIPTION
**Pull request by mathxlee**
1. fix: get the wrong campaign url part
2. feat: claim OAT feature

### CODE REVIEW

File: src/plugins/projectGalaxy/index.ts [hunk 1]
```diff
@@ -35,6 +39,30 @@ export default class Plugin {
...
+  async claim(address, campaignID) {
+    const eventResponse = await this.fetchGQL({
+      query: `mutation claimOAT($input: PrepareParticipateInput!) {
+        prepareParticipate(input: $input) {
+          allow
+        }
+      }`,
+      variables: {
+        input: {
+          signature: '',
```
Comment (mktcode) [inline src/plugins/projectGalaxy/index.ts:51]: This empty signature is intentional?

File: src/plugins/projectGalaxy/components/CustomBlock.vue [hunk 2]
```diff
@@ -129,7 +129,8 @@ export default {
         this.oats = this.space.plugins.projectGalaxy.oats;
         this.currentCampaignUrl = this.oats[this.proposal.id];
         if (this.currentCampaignUrl) {
-          this.currentCampaignId = this.currentCampaignUrl.match(/[^/]\w+/g)[2];
+          this.currentCampaignId =
+            this.currentCampaignUrl.match(/[^/]+(?=\/$|$)/g)[0];
```
Comment (mktcode) [inline src/plugins/projectGalaxy/components/CustomBlock.vue:133]: I love regex but I also love readability. A simple `split('/').pop()` would've done the trick here I guess. 😃

File: src/plugins/projectGalaxy/index.ts (hunk 1 above)
Comment (mathxlee) [reply src/plugins/projectGalaxy/index.ts:51]: yes, it is necessary

File: src/plugins/projectGalaxy/components/CustomBlock.vue (hunk 2 above)
Comment (mathxlee) [reply src/plugins/projectGalaxy/components/CustomBlock.vue:133]: yeah, `split('/').pop()` will be more elegant

Comment (mathxlee) [reply src/plugins/projectGalaxy/components/CustomBlock.vue:133]: but maybe next time we will do the optimization?

**Pull request merged**